### ISSUE TITLE
[help] Problems using proxy

### ISSUE THREAD

**Issue by lucasmsoares96** [JuliaWeb/HTTP.jl#899]
I'm having problems making a request using proxy. When I make the request without the proxy everything works fine.
```julia
using HTTP
resp2 = HTTP.get(
    "https://api.ipify.org?format=json",
)
```
But when I add the proxy:
```julia
using HTTP
resp2 = HTTP.get(
    "https://api.ipify.org?format=json",
    proxy = "http://203.30.189.192:80"
)
```
I get the following error 
```
ERROR: HTTP.ExceptionRequest.StatusError(400, "CONNECT", "api.ipify.org:443", HTTP.Messages.Response:
"""
HTTP/1.1 400 Bad Request
Server: cloudflare
Date: Sun, 07 Aug 2022 23:29:33 GMT
Content-Type: text/html
Content-Length: 155
Connection: close
CF-RAY: -

""")
Stacktrace:
  [1] request(::Type{ExceptionLayer{ConnectionPoolLayer{StreamLayer{Union{}}}}}, ::URIs.URI, ::Vararg{Any}; kw::Base.Pairs{Symbol, Any, Tuple{Symbol, Symbol, Symbol}, NamedTuple{(:iofunction, :reached_redirect_limit, :proxy), Tuple{Nothing, Bool, String}}})
    @ HTTP.ExceptionRequest ~/.julia/packages/HTTP/aTjcj/src/ExceptionRequest.jl:22
  [2] (::Base.var"#76#78"{Base.var"#76#77#79"{ExponentialBackOff, HTTP.RetryRequest.var"#2#3"{Bool, HTTP.Messages.Request}, typeof(HTTP.request)}})(::Type, ::Vararg{Any}; kwargs::Base.Pairs{Symbol, Any, Tuple{Symbol, Symbol, Symbol}, NamedTuple{(:iofunction, :reached_redirect_limit, :proxy), Tuple{Nothing, Bool, String}}})
    @ Base ./error.jl:294
  [3] #request#1
    @ ~/.julia/packages/HTTP/aTjcj/src/RetryRequest.jl:44 [inlined]
  [4] request(::Type{MessageLayer{RetryLayer{ExceptionLayer{ConnectionPoolLayer{StreamLayer{Union{}}}}}}}, method::String, url::URIs.URI, headers::Vector{Pair{SubString{String}, SubString{String}}}, body::Vector{UInt8}; http_version::VersionNumber, target::String, parent::Nothing, iofunction::Nothing, kw::Base.Pairs{Symbol, Any, Tuple{Symbol, Symbol}, NamedTuple{(:reached_redirect_limit, :proxy), Tuple{Bool, String}}})
    @ HTTP.MessageRequest ~/.julia/packages/HTTP/aTjcj/src/MessageRequest.jl:66
  [5] request(::Type{BasicAuthLayer{MessageLayer{RetryLayer{ExceptionLayer{ConnectionPoolLayer{StreamLayer{Union{}}}}}}}}, method::String, url::URIs.URI, headers::Vector{Pair{SubString{String}, SubString{String}}}, body::Vector{UInt8}; kw::Base.Pairs{Symbol, Any, Tuple{Symbol, Symbol}, NamedTuple{(:reached_redirect_limit, :proxy), Tuple{Bool, String}}})
    @ HTTP.BasicAuthRequest ~/.julia/packages/HTTP/aTjcj/src/BasicAuthRequest.jl:28
  [6] request(::Type{RedirectLayer{BasicAuthLayer{MessageLayer{RetryLayer{ExceptionLayer{ConnectionPoolLayer{StreamLayer{Union{}}}}}}}}}, method::String, url::URIs.URI, headers::Vector{Pair{SubString{String}, SubString{String}}}, body::Vector{UInt8}; redirect_limit::Int64, forwardheaders::Bool, kw::Base.Pairs{Symbol, String, Tuple{Symbol}, NamedTuple{(:proxy,), Tuple{String}}})
    @ HTTP.RedirectRequest ~/.julia/packages/HTTP/aTjcj/src/RedirectRequest.jl:28
  [7] #request#1
    @ ~/.julia/packages/HTTP/aTjcj/src/TopRequest.jl:15 [inlined]
  [8] request(method::String, url::String, h::Vector{Pair{SubString{String}, SubString{String}}}, b::Vector{UInt8}; headers::Vector{Pair{SubString{String}, SubString{String}}}, body::Vector{UInt8}, query::Nothing, kw::Base.Pairs{Symbol, String, Tuple{Symbol}, NamedTuple{(:proxy,), Tuple{String}}})
    @ HTTP ~/.julia/packages/HTTP/aTjcj/src/HTTP.jl:330
  [9] #get#17
    @ ~/.julia/packages/HTTP/aTjcj/src/HTTP.jl:407 [inlined]
 [10] top-level scope
    @ ~/Linkedin/proxy.jl:42
```
I tested the ip in Python and everything worked fine:
```python
import requests
proxies = {
    'http': 'http://203.30.189.192:80',
}
url = 'https://api.ipify.org?format=json'
requests.get(url, proxies=proxies).text
 ```
any tips on how to solve this?

 - Julia 1.7.3
 - HTTP.jl 0.9.17

**Comment by quinnj** [comment]
Can you update to the latest HTTP.jl release? (1.2.0 I think?) There have been a few proxy-related fixes since the 0.9.17 release.

**Comment by lucasmsoares96** [comment]
@quinnj  That's right. The MLJ package was preventing HTTP.jl from updating to the latest version. Thank you very much!
